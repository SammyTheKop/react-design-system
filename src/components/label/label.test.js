import React from "react";

// component
import Label from "./label";

// jest renderer
import renderer from "react-test-renderer";

// testing library react
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

// Case 1: Label Renders properly or not
it("Case 1: Label Component Renders Correctly", () => {
  const labelComponentTree = renderer.create(<Label>Test</Label>).toJSON();
  expect(labelComponentTree).toMatchSnapshot();
});

// Case 2: Label Component renders the correct value or not
it("Case 2: Label Component renders the correct value or not (Positive Testing", () => {
  const { getByTestId } = render(<Label>Test</Label>);
  expect(getByTestId("label").textContent).toEqual("Test");
});

// Case 3: Label Component renders the correct value or not
it("Case 3: Label Component renders the correct value or not (Negative Testing", () => {
  const { getByTestId } = render(<Label>Test - 1</Label>);
  expect(getByTestId("label").textContent).not.toEqual("Test");
});

// Case 4: Label Component test id is present or not
it("Case 4: Label Component test is is present orr not", () => {
  const { getByTestId } = render(<Label>Test - 1</Label>);
  expect(getByTestId("label")).toBeInTheDocument();
});
