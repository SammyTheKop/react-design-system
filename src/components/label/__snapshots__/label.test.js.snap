// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Case 1: Label Component Renders Correctly 1`] = `
Array [
  <div
    className="label-component"
  >
    <label
      className="label medium"
      data-testid="label"
      htmlFor=""
    >
      Test
    </label>
  </div>,
  <style
    jsx={true}
  >
    
          /* Styles for the Label Component */
          /* Keep adding the Styles of the Label Size Props below and other CSS properties above */
          .label.extra-large {
            font-size: 24px;
          }

          .label.large {
            font-size: 20px;
          }

          .label.medium {
            font-size: 16px;
          }

          .label.small {
            font-size: 14px;
          }

          .label.extra-small {
            font-size: 10px;
          }
        
  </style>,
]
`;
